x/profiles/keeper/relay_ibc.go

```diff
@@ -0,0 +1,87 @@
+package keeper
+
+import (
+	"errors"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	channeltypes "github.com/cosmos/cosmos-sdk/x/ibc/core/04-channel/types"
+	"github.com/desmos-labs/desmos/x/profiles/types"
+)
+
+// OnRecvPacket processes packet reception
+func (k Keeper) OnRecvPacket(
```

Comment:
Most of the logic present here can be simply moved inside the `StoreChainLink` method as mentioned above as well. This would reduce the code duplication and make the code more easily maintainable